zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -0,0 +1,324 @@
+      SSL_X509_CONFIG_PREFIX + "clientCertIdSanExtractMatcherGroupIndex";
+  public static final String SUBJECT_ALTERNATIVE_NAME_SHORT = "SAN";
+  private static final String DEFAULT_REGEX = ".*";
+  private String clientCertIdType;
+  private int clientCertIdSanMatchType = -1;
+  private String clientCertIdSanMatchRegex;
+  private String clientCertIdSanExtractRegex;
+  private int clientCertIdSanExtractMatcherGroupIndex = -1;
+
+  /** ZooKeeper server-side ZNode group ACL feature */
+//   x509ClientIdAsAclEnabled enables/disables whether znodes created by auth'ed clients
+//   should have ACL fields populated with the client Id given by the authentication provider.
+//   Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
+  public static final String ZNODE_GROUP_ACL_CONFIG_PREFIX = "zookeeper.X509ZNodeGroupAclProvider.";
+  // Enables/disables whether znodes created by auth'ed clients
+  // should have ACL fields populated with the client Id given by the authentication provider.
+  // Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
+  public static final String SET_X509_CLIENT_ID_AS_ACL =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "setX509ClientIdAsAcl";
+  // A list of domain names that will have cross-domain access privilege, separated by ","
+  public static final String SUPER_USER_DOMAIN_NAME =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "superUserDomainName";
+  // A defined URI represents a super user
+  public static final String ZOOKEEPER_ZNODEGROUPACL_SUPERUSER_ID =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "superUserId";
+  // A list of znode path prefixes, separated by ","
+  // Znode whose path starts with the defined path prefix would have open read access
+  // Meaning the znode will have (world:anyone, r) ACL
+  public static final String OPEN_READ_ACCESS_PATH_PREFIX =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "openReadAccessPathPrefix";
+  // If the server is dedicated for one domain, use this config property to define the domain name,
+  // and enable connection filtering feature for this domain
+  public static final String DEDICATED_DOMAIN = ZNODE_GROUP_ACL_CONFIG_PREFIX + "dedicatedDomain";
+  public static final String CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "clientUriDomainMappingRootPath";
+
+  private static boolean x509ClientIdAsAclEnabled;
+  private String znodeGroupAclSuperUserId;
+  private String znodeGroupAclSuperUserDomainNameStr;
+  private String znodeGroupAclOpenReadAccessPathPrefixStr;
+  private String znodeGroupAclServerDedicatedDomain;
+  private String znodeGRoupAclClientUriDomainMappingRootPath;
+  // Although using "volatile" keyword with double checked locking could prevent the undesired
+  //creation of multiple objects; not using here for the consideration of read performance
+  private Set<String> openReadAccessPathPrefixes;
+  private Set<String> superUserDomainNames;
```

Comment:
"superUserDomain" -> ?

Reply:
There will be multiple domains, so I think you meant `superUserDomains`? Also, according to previous discussions, should we change this to `crossDomainDomains`? This name is a bit strange IMO. Shall we keep `superUserDomains`, and just clarify what "superUser" means under this context?

Reply:
Sorry, what I meant here is, do we want to find a better name than "superUser"DomainNames? 
Maybe something like `crossDomainAccessDomainNames`/`multiDomainAccessDomainNames`?